inspire_crawler/tasks.py

```diff
@@ -75,10 +75,10 @@ def _extract_results_data(results_path):
         )
         results_data = []
         with open(results_path) as records:
-            lines = [
-                line.strip() for line in records.readlines()
-                if line.strip()
-            ]
+            lines = (
+                line.strip() for line in records if line.strip()
+            )
+
```

Comment:
Can't you just:
```python
for line in records:
    line = line.strip()
    if not line:
        continue
```

With the current code you are still loading everything into memory.